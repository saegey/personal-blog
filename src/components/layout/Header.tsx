import { Button, Box, Flex } from 'theme-ui'

type Props = {
  setMenuOpen: (arg: boolean) => void
}

const Header = ({ setMenuOpen }: Props) => {
  return (
    <header>
      <Box
        sx={{
          // position: 'fixed',
          zIndex: 3,
          marginLeft: '20px',
          marginTop: '10px',
        }}
      >
        <Button
          title="Toggle Menu"
          sx={{
            appearance: 'none',
            bg: 'cardBackground',
            width: 32,
            height: 32,
            m: 0,
            p: 0,
            color: 'inherit',
            background: 'none',
            // bg: 'transparent',
            border: 0,
            // ':focus': {
            //   outline: '1px solid',
            // },
            ':hover': {
              color: 'primary',
            },
            // boxShadow: '0px 2px 5px rgba(0, 0, 0, 0.2)',
          }}
          onClick={() => {
            setMenuOpen(true)
          }}
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="20"
            height="20"
            viewBox="0 0 100 80"
            sx={{
              display: 'block',
              margin: 0,
              fill: 'text',
            }}
          >
            <rect width="100" height="15"></rect>
            <rect y="30" width="100" height="15"></rect>
            <rect y="60" width="100" height="15"></rect>
          </svg>
        </Button>
      </Box>
      <Flex
        sx={{
          zIndex: 3,
          alignItems: 'center',
          background: 'headerColor',
          top: 0,
        }}
      >
        <Box sx={{ flexGrow: 1 }}></Box>
        <Box
          sx={{
            // display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
          }}
        >
          <div
            sx={{
              width: ['100px', '110px', '130px'],
            }}
          >
            {/* <Logo border={false} /> */}
          </div>
        </Box>
        <Box sx={{ flexGrow: 1 }}></Box>
      </Flex>
    </header>
  )
}

export default Header
