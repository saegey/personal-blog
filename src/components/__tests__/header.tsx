/**
 * @jest-environment jsdom
 */

// import React from 'react'
import renderer from 'react-test-renderer'
// import { render } from '@testing-library/react'

import { ThemeProvider } from 'theme-ui'

import Header from '../header'

import { theme } from './__test-utils__'

describe('Header', () => {
  it('renders', () => {
    const json = renderer.create(
        <ThemeProvider theme={theme}>
          <Header setMenuOpen={jest.fn()}/>
        </ThemeProvider>
      ).toJSON()
    expect(json).toMatchSnapshot()
  })
})
