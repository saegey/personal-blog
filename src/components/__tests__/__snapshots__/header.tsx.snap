// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  z-index: 3;
  margin-left: 20px;
  margin-top: 10px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  color: var(--theme-ui-colors-background);
  background-color: var(--theme-ui-colors-primary);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background-color: cardBackground;
  width: 32px;
  height: 32px;
  margin: 0;
  padding: 0;
  color: inherit;
  background: none;
  border: 0;
}

.emotion-2:hover {
  color: var(--theme-ui-colors-primary);
}

.emotion-3 {
  display: block;
  margin: 0;
  fill: text;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  z-index: 3;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: var(--theme-ui-colors-headerColor);
  top: 0;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-7 {
  width: 100px;
}

@media screen and (min-width: 40em) {
  .emotion-7 {
    width: 110px;
  }
}

@media screen and (min-width: 52em) {
  .emotion-7 {
    width: 130px;
  }
}

<header
  className="emotion-0"
>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-2"
      onClick={[Function]}
      title="Toggle Menu"
    >
      <svg
        className="emotion-3"
        height="20"
        viewBox="0 0 100 80"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          height="15"
          width="100"
        />
        <rect
          height="15"
          width="100"
          y="30"
        />
        <rect
          height="15"
          width="100"
          y="60"
        />
      </svg>
    </button>
  </div>
  <div
    className="emotion-4"
  >
    <div
      className="emotion-5"
    />
    <div
      className="emotion-6"
    >
      <div
        className="emotion-7"
      />
    </div>
    <div
      className="emotion-5"
    />
  </div>
</header>
`;
